astroplan/tests/test_core.py

```diff
@@ -155,3 +152,591 @@ def test_FixedTarget_ra_dec():
+                                                         which='nearest').datetime
+
+    # Computed in print_pyephem_solar_transit_noon()
+    pyephem_next_transit = datetime.datetime(2000, 1, 1, 12, 3, 17, 207300)
+    pyephem_next_antitransit = datetime.datetime(2000, 1, 2, 0, 3, 31, 423333)
+    pyephem_prev_transit = datetime.datetime(1999, 12, 31, 12, 2, 48, 562755)
+    pyephem_prev_antitransit = datetime.datetime(2000, 1, 1, 0, 3, 2, 918943)
+
+    threshold_minutes = 8
+    assert (abs(astroplan_next_transit - pyephem_next_transit) <
+            datetime.timedelta(minutes=threshold_minutes))
+    assert (abs(astroplan_next_antitransit - pyephem_next_antitransit) <
+            datetime.timedelta(minutes=threshold_minutes))
+    assert (abs(astroplan_prev_transit - pyephem_prev_transit) <
+            datetime.timedelta(minutes=threshold_minutes))
+    assert (abs(astroplan_prev_antitransit - pyephem_prev_antitransit) <
+            datetime.timedelta(minutes=threshold_minutes))
+
+    # Check nearest
+    assert astroplan_next_transit == astroplan_nearest_transit
+    assert astroplan_nearest_antitransit == astroplan_prev_antitransit
+
+def test_solar_transit_convenience_methods():
+    '''
+    Test that astroplan's noon and midnight convenience methods agree with
+    PyEphem's solar transit/antitransit time.
+    '''
+    lat = '00:00:00'
+    lon = '00:00:00'
+    elevation = 0.0 * u.m
+    pressure = 0 * u.bar
+    location = EarthLocation.from_geodetic(lon, lat, elevation)
+    time = Time('2000-01-01 12:00:00')
+    from astropy.coordinates import get_sun
+    obs = Observer(location=location, pressure=pressure)
+
+    # Compute next/previous noon/midnight using generic calc_transit methods
+    astroplan_next_noon = obs.noon(time, which='next').datetime
+    astroplan_next_midnight = obs.midnight(time, which='next').datetime
+    astroplan_prev_noon = obs.noon(time, which='previous').datetime
+    astroplan_prev_midnight = obs.midnight(time, which='previous').datetime
+
+    # Computed in print_pyephem_solar_transit_noon()
+    pyephem_next_transit = datetime.datetime(2000, 1, 1, 12, 3, 17, 207300)
+    pyephem_next_antitransit = datetime.datetime(2000, 1, 2, 0, 3, 31, 423333)
+    pyephem_prev_transit = datetime.datetime(1999, 12, 31, 12, 2, 48, 562755)
+    pyephem_prev_antitransit = datetime.datetime(2000, 1, 1, 0, 3, 2, 918943)
+
+    threshold_minutes = 8
+    assert (abs(astroplan_next_noon - pyephem_next_transit) <
+            datetime.timedelta(minutes=threshold_minutes))
+    assert (abs(astroplan_next_midnight - pyephem_next_antitransit) <
+            datetime.timedelta(minutes=threshold_minutes))
+    assert (abs(astroplan_prev_noon - pyephem_prev_transit) <
+            datetime.timedelta(minutes=threshold_minutes))
+    assert (abs(astroplan_prev_midnight - pyephem_prev_antitransit) <
+            datetime.timedelta(minutes=threshold_minutes))
+
+def print_pyephem_solar_transit_noon():
+    '''
```

Comment:
I'm not sure it's really needed, but if you add this to the docs it should be on a separate page for developers, not mixed in with docs for end-users.

Example how I did this in Gammapy ... I used a Q & A format: https://gammapy.readthedocs.org/en/latest/development/index.html

Astropy has this: http://astropy.readthedocs.org/en/latest/#developer-documentation
I'm not sure it's a useful example for such a small package as astroplan.